src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -3045,6 +3045,148 @@ public static void XmlMembersMapping_CompositeType()
+</DefaultValuesSetToNaN>");
+        Assert.NotNull(actual);
+        Assert.Equal(actual, value);
+    }
+
+    [Fact]
+    public static void Xml_NullRefInXmlSerializerCtorTest()
+    {
+        string defaultNamespace = "http://www.contoso.com";
+        var value = PurchaseOrder.CreateInstance();
+        string baseline = 
+@"<?xml version=""1.0""?>
+<PurchaseOrder xmlns:xsd=""http://www.w3.org/2001/XMLSchema"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns=""http://www.cpandl.com"">
+  <ShipTo Name=""John Doe"">
+    <Line1>1 Main St.</Line1>
+    <City>AnyTown</City>
+    <State>WA</State>
+    <Zip>00000</Zip>
+  </ShipTo>
+  <OrderDate>Monday, April 10, 2017</OrderDate>
+  <Items>
+    <OrderedItem>
+      <ItemName>Widget S</ItemName>
+      <Description>Small widget</Description>
+      <UnitPrice>5.23</UnitPrice>
+      <Quantity>3</Quantity>
+      <LineTotal>15.69</LineTotal>
+    </OrderedItem>
+  </Items>
+  <SubTotal>15.69</SubTotal>
+  <ShipCost>12.51</ShipCost>
+  <TotalCost>28.20</TotalCost>
+</PurchaseOrder>";
+
+        var actual = SerializeAndDeserialize(value,
+            baseline,
+            () => new XmlSerializer(value.GetType(), null, null, null, defaultNamespace)
+            );
+        Assert.NotNull(actual);
+
+        actual = SerializeAndDeserialize(value,
+            baseline,
+            () => new XmlSerializer(value.GetType(), null, null, null, defaultNamespace, null)
+            );
+        Assert.NotNull(actual);
+    }
+
+    [Fact]
+    public static void Xml_AliasedPropertyTest()
+    {
+        var inputList = new List<string> { "item0", "item1", "item2", "item3", "item4" };
+        var value = new AliasedTestType { Aliased = inputList };
+        var actual = SerializeAndDeserialize(value,
+@"<?xml version=""1.0""?>
+<AliasedTestType xmlns:xsd=""http://www.w3.org/2001/XMLSchema"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"">
+  <Y>
+    <string>item0</string>
+    <string>item1</string>
+    <string>item2</string>
+    <string>item3</string>
+    <string>item4</string>
+  </Y>
+</AliasedTestType>");
+
+        Assert.NotNull(actual);
+        Assert.NotNull(actual.Aliased);
+        Assert.Equal(actual.Aliased.GetType(), inputList.GetType());
+        Assert.Equal(((List<string>)actual.Aliased).Count, inputList.Count);
+        for (int i = 0; i < inputList.Count; i++)
+        {
+            Assert.Equal(((List<string>)actual.Aliased).ElementAt(i), inputList[i]);
+        }
+    }
+
+    [Fact]
+    public static void Xml_DeserializeHiddenMembersTest()
+    {
+        var xmlSerializer = new XmlSerializer(typeof(DerivedClass1));
+        string inputXml = "<DerivedClass1><Prop>2012-07-07T00:18:29.7538612Z</Prop></DerivedClass1>";
+
+        using (var reader = new StringReader(inputXml))
+        {
+            var derivedClassInstance = (DerivedClass1)xmlSerializer.Deserialize(reader);
+            Assert.NotNull(derivedClassInstance.Prop);
```

Comment:
All right, will do.  It was thoughtless of me when porting the test :( thank you for all the comments!